plugins/Toolbox/src/Toolbox.py

```diff
@@ -0,0 +1,613 @@
+
+    @pyqtSlot()
+    def restart(self):
+        self._package_manager._removeAllScheduledPackages()
+        CuraApplication.getInstance().windowClosed()
+
+
+
+    # Checks
+    # --------------------------------------------------------------------------
+    @pyqtSlot(str, result = bool)
+    def canUpdate(self, package_id: str) -> bool:
+        local_package = self._package_manager.getInstalledPackageInfo(package_id)
+        if local_package is None:
+            return False
+
+        remote_package = None
+        for package in self._metadata["packages"]:
+            if package["package_id"] == package_id:
+                remote_package = package
+        if remote_package is None:
+            return False
+
+        local_version = local_package["package_version"]
+        remote_version = remote_package["package_version"]
+        return Version(remote_version) > Version(local_version)
+
+    @pyqtSlot(str, result = bool)
+    def isInstalled(self, package_id: str) -> bool:
+        return self._package_manager.isPackageInstalled(package_id)
+
+    @pyqtSlot(str, result = bool)
+    def isEnabled(self, package_id: str) -> bool:
+        if package_id in self._plugin_registry.getActivePlugins():
+            return True
+        return False
+
+
+
+    # Make API Calls
+    # --------------------------------------------------------------------------
+    def _makeRequestByType(self, type: str):
+        Logger.log("i", "Toolbox: Requesting %s metadata from server.", type)
+        request = QNetworkRequest(self._request_urls[type])
+        request.setRawHeader(*self._request_header)
+        self._network_manager.get(request)
+    # TODO: Request authors and request material showcase
+
+    @pyqtSlot(str)
+    def startDownload(self, url: str):
+        Logger.log("i", "Toolbox: Attempting to download & install package from %s.", url)
+        url = QUrl(url)
+        self._download_request = QNetworkRequest(url)
+        self._download_request.setAttribute(QNetworkRequest.RedirectPolicyAttribute, QNetworkRequest.NoLessSafeRedirectPolicy)
+        self._download_request.setRawHeader(*self._request_header)
+        self._download_reply = self._network_manager.get(self._download_request)
+        self.setDownloadProgress(0)
+        self.setIsDownloading(True)
+        self._download_reply.downloadProgress.connect(self._onDownloadProgress)
+
+    @pyqtSlot()
+    def cancelDownload(self):
+        Logger.log("i", "Toolbox: User cancelled the download of a plugin.")
+        self._download_reply.abort()
+        self._download_reply.downloadProgress.disconnect(self._onDownloadProgress)
+        self._download_reply = None
+        self._download_request = None
+        self.setDownloadProgress(0)
+        self.setIsDownloading(False)
+
+
+
+    # Handlers for Network Events
+    # --------------------------------------------------------------------------
+    def _onNetworkAccesibleChanged(self, accessible: int):
```

Comment:
Typo in function name, should be accessible.

Reply:
Fixed in ec8b955